aspnetcore/mvc/models/file-uploads/samples/2.x/SampleApp/Utilities/FileHelpers.cs

```diff
@@ -0,0 +1,261 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Net.Http.Headers;
+
+namespace SampleApp.Utilities
+{
+    public static class FileHelpers
+    {
+        // If you require a check on specific characters in the IsValidFileExtensionAndSignature
+        // method, supply the characters in the _allowedChars field.
+        private static readonly byte[] _allowedChars = { };
+        // For more file signatures, see the File Signatures Database: https://www.filesignatures.net/
+        private static readonly Dictionary<string, byte[]> _fileSignature = new Dictionary<string, byte[]>
+        {
+            { ".doc", new byte[] { 0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1 } },
+            { ".docx", new byte[] { 0x50, 0x4B, 0x03, 0x04 } },
+            { ".gif", new byte[] { 0x47, 0x49, 0x46, 0x38 } },
+            { ".jpeg", new byte[] { 0xFF, 0xD8, 0xFF, 0xE0, 0xFF, 0xD8, 0xFF, 0xE2, 0xFF, 0xD8, 0xFF, 0xE3 } },
+            { ".jpg", new byte[] { 0xFF, 0xD8, 0xFF, 0xE0, 0xFF, 0xD8, 0xFF, 0xE1, 0xFF, 0xD8, 0xFF, 0xE8 } },
+            { ".png", new byte[] { 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A } },
+            { ".pdf", new byte[] { 0x25, 0x50, 0x44, 0x46 } },
+            { ".ppt", new byte[] { 0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1 } },
+            { ".pptx", new byte[] { 0x50, 0x4B, 0x03, 0x04 } },
+            { ".xls", new byte[] { 0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1 } },
+            { ".xlsx", new byte[] { 0x50, 0x4B, 0x03, 0x04 } },
+            { ".zip", new byte[] { 0x50, 0x4B, 0x03, 0x04, 0x50, 0x4B, 0x4C, 0x49, 0x54, 0x55, 0x50, 0x4B, 0x53, 0x70, 0x58, 0x50, 0x4B, 0x05, 0x06, 0x50, 0x4B, 0x07, 0x08, 0x57, 0x69, 0x6E, 0x5A, 0x69, 0x70 } },
```

Comment:
I think these are several possible values for zip files. I don't think the header is this large.

Reply:
That's what the Azure engineer recommended.

https://docs.microsoft.com/en-us/azure/security/develop/threat-modeling-tool-input-validation#example-8

Do we want to track that engineer down and see if they can shed more light on it?

Reply:
@guardrex @pranavkm there are different version on which you can check. But as far as I know the most common is PKZIP 1 which is `0x50 0x4B 0x03 0x04`. You also have then indication of what type zip it is. Eg. `0x50 0x4B 0x05 0x06` is empty archive and `0x50 0x4B 0x07 0x08` multi volume archive file.  :)

Reply:
With the new bits on this morning's commit, I put all of the ZIP sigs in there from the File Signatures Database. See how that implementation looks with the fancy one-liner check.